### PR TITLE
[i18n/audio] Improve audio feedback for candidate selection/deselection.

### PR DESCRIPTION
**Pull request by kofi-q**
## Overview

Add a "votes remaining in this contest" audio readout after a voter selects or deselects a candidate. This provides audio-only votes feature parity with the corresponding "votes remaining" status  rendered at the top of the contest page.

## Demo Video or Screenshot [🔊]

https://github.com/votingworks/vxsuite/assets/264902/3f5cc03d-93ef-4abe-9888-87d6cc2bd1ac

## Testing Plan
- Additional test assertions for the new behaviour
- Manual run with test audio data

## Checklist

- ~[ ] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.~

- [x] I have added a screenshot and/or video to this PR to demo the change
- [x] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

### CODE REVIEW

File: libs/mark-flow-ui/src/components/candidate_contest.tsx [hunk 1]
```diff
@@ -92,30 +92,43 @@ export function CandidateContest({
...
       return () => clearTimeout(timer);
     }
-  }, [deselectedCandidate]);
+  }, [recentlyDeselectedCandidate]);
+
+  useEffect(() => {
+    if (recentlySelectedCandidate !== '') {
+      const timer = setTimeout(() => {
+        setRecentlySelectedCandidate('');
+      }, 100);
```
Comment (arsalansufi) [inline libs/mark-flow-ui/src/components/candidate_contest.tsx:115]: If the `<AudioOnly>` element is removed while the audio is still being read out, the audio will still be allowed to finish, is that right? I imagine so given the video in the PR description - just double checking

**Review by arsalansufi — APPROVED** [review]
Nice!

File: libs/mark-flow-ui/src/components/candidate_contest.tsx (hunk 1 above)
Comment (kofi-q) [reply libs/mark-flow-ui/src/components/candidate_contest.tsx:115]: Yeah, good question - the screen reader does check to see if the element's still attached to the DOM before playing audio, but only at the beginning, so this just allows enough time for the audio to start, at which point the element can be removed without affecting playback

Comment (arsalansufi) [reply libs/mark-flow-ui/src/components/candidate_contest.tsx:115]: Got it, thanks for explaining!

**Pull request merged**